### PR TITLE
System property extension

### LINKED ISSUE #549: Properties files with .sl extension
rethink how to distinguish between SP and executable .sl files (now it's in CLI / builder naive implementation)

### PR DESCRIPTION
**Pull request by Bonczidai**
fixes #549 
related to #321 
waiting for #553 - done

Signed-off-by: Bonczidai Levente <levente.bonczidai@hpe.com>

### CODE REVIEW

**Comment by Bonczidai** [comment]
changed the SP extension, @oritstone please take a look

File: cloudslang-content-verifier/src/main/java/io/cloudslang/lang/tools/build/verifier/SlangContentVerifier.java [hunk 1]
```diff
@@ -40,6 +34,7 @@
     private final static Logger log = Logger.getLogger(SlangContentVerifier.class);
 
     private String[] SLANG_FILE_EXTENSIONS = {"sl", "sl.yaml", "sl.yml"};
+    private static final String[] PROPERTIES_FILE_EXTENSIONS = {"prop.sl"};
```
Comment (oritstone) [inline cloudslang-content-verifier/src/main/java/io/cloudslang/lang/tools/build/verifier/SlangContentVerifier.java]: Why do you need to hold this constant twice (also in TestCasesYamlParser)?
You should hold it once and refer to the same constant

File: cloudslang-content-verifier/src/main/java/io/cloudslang/lang/tools/build/verifier/SlangContentVerifier.java [hunk 2]
```diff
@@ -47,37 +42,28 @@
...
+            try {
+                sourceModel = slangCompiler.preCompile(SlangSource.fromFile(slangFile));
+            } catch (Exception e) {
+                String errorMessage = "Failed creating Slang models for file: \'" + slangFile.getAbsoluteFile() + "\'.\n" + e.getMessage();
+                log.error(errorMessage);
+                throw new RuntimeException(errorMessage, e);
+            }
+            if (sourceModel != null) {
+                staticSlangFileValidation(slangFile, sourceModel);
+                slangModels.put(getUniqueName(sourceModel), sourceModel);
```
Comment (oritstone) [inline cloudslang-content-verifier/src/main/java/io/cloudslang/lang/tools/build/verifier/SlangContentVerifier.java]: The duplication of code with the CLI keeps on growing...
I think we should deal with the common module refactor task already...

**Comment by oritstone** [comment]
:+1: 
You can merge after fixing the minor issue

**Comment by Bonczidai** [comment]
thanks @oritstone  - merging, indeed we should deal with the common module stuff, the sooner the better

**Pull request merged**